src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,316 @@
+import datetime
+import xml.etree.ElementTree as ET
+import os
+import zipfile
+import glob
+import numpy as np
+
+
+
+class annotation:
+    # A parent class for annotation reader for Calibrarion, Noise, and poteltially L1
+    def __init__(self,path_annotation:str,polarization='VH',subswath=1):
+        #Attributes
+        self._IPF_version=None
+        self.path_annotation=path_annotation #Accepted format: SAFE .zip file, SAFE directory, or annotation .xml file
+        self.polarization=polarization
+        self.subswath=subswath #Subswath of the annotation file to load
+        self._xml_loaded=None
+        self.kind=''
+
+        #Data access point for the annotation .xml file
+        self.xml_et=None
+
+    def _get_search_phrase(self):
+        str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+        token_safe=str_safe.split('_')
+
+        phrase_base='{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+
+        if self.kind=='calibration':
+            phrase_xml_to_search='calibration-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='noise':
+            phrase_xml_to_search='noise-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='':
+            phrase_xml_to_search=f'/{phrase_base}'
+        else:
+            raise ValueError(f'Cannot recognize the annotation type {self.kind}')
+
+        return phrase_xml_to_search
+
+
+    def _load_xml_as_et(self): #kind: ['', 'calibration', 'noise']
+        if os.path.isfile(self.path_annotation):
+            if self.path_annotation.endswith('.zip'):
+                zip_in=zipfile.ZipFile(self.path_annotation,'r')
```

Comment:
I think you could use `with` here:
```suggestion
                with zipfile.ZipFile(self.path_annotation,'r') as zip_in:
```
If you use that, you can remove `zip_in.close()`

Reply:
That's a good suttestion. Thanks!
https://github.com/seongsujeong/s1-reader/blob/7e85dd7cda8d2ff3d36779894b9e0c52365d6149/src/s1reader/s1_annotation.py#L59-L76